main.cpp

```diff
@@ -36,60 +36,47 @@ T::T(int v, const char* p): value(v), name(p) {}
 
 struct Comparator                               //4
 {
-    T* compare(T* a, T* b) //5
+    T* compare(T& a, T& b) //5
     {
-        if (a != nullptr && b != nullptr)
-        {
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
-        }        
+        if( a.value < b.value ) return &a;
+        if( a.value > b.value ) return &b;     
         return nullptr;
     }
 };
 
 struct U
 {
     float x { 0 }, y{ 0 };
-    float computeDistance(float* xUpdated)      //12
+    float computeDistance(const float &xUpdated)      //12
     {
-        if (xUpdated != nullptr)
+        std::cout << "U's x value: " << this->x << std::endl;
+        this->x = xUpdated;
+        std::cout << "U's y updated value: " << this->x << std::endl;
+        while( std::abs( this->y - this->x ) > 0.001f )
         {
-            std::cout << "U's x value: " << this->x << std::endl;
-            this->x = *xUpdated;
-            std::cout << "U's y updated value: " << this->x << std::endl;
-            while( std::abs(this->y - this->x) > 0.001f )
-            {
-                this->y += .1f;
-            }
-            std::cout << "U's computeDistance updated value: " << this->y << std::endl;
-            return this->y * this->x;
+            this->y += .1f;
         }
-        std::cout << "Not allocated pointer " << std::endl;
-        return 0;
+        std::cout << "U's computeDistance updated value: " << this->y << std::endl;
+        return this->y * this->x;
     }
 };
 
 struct StaticU
 {
-    static float computeDistance(U* that, float* xUpdated )        //10
+    static float computeDistance(U &that, const float &xUpdated )        //10
```

Comment:
put the `&` next to the type.   when you put it next the variable name, it looks like you're trying to get the address of the variable.
- [ ] revise